hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/RecordCacheImpl.java

```diff
@@ -250,23 +311,17 @@ public List<TransactionRecord> getRecords(@NonNull final AccountID accountID) {
         for (final var transactionID : transactionIDs) {
             final var history = histories.get(transactionID);
             if (history != null) {
-                final var recs = history.records();
+                final var recs = history.orderedRecords();
                 records.addAll(recs.size() > maxRemaining ? recs.subList(0, maxRemaining) : recs);
                 maxRemaining -= recs.size();
                 if (maxRemaining <= 0) break;
```

Comment:
Not sure if it matters, but these may not be fully ordered if there are more than the maximum limit of transactions to be returned.
The transactions never added to the `records` list could have been anywhere in the final ordering.
If we promise any kind of final ordering in the return value, then we would have to build the **full** list, sort it, *then* trim to length.